.github/workflows/trivy-branch.yaml

```diff
@@ -0,0 +1,50 @@
+# trivy-branch.yaml
+# Trivy: Scan of current branch
+# Trivy is a comprehensive and versatile security scanner.
+# Trivy has scanners that look for security issues,
+# and targets where it can find those issues.
+# https://github.com/aquasecurity/trivy
+---
+name: Trivy - branch scan
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+
+concurrency:
+  group: trivy-${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  scan:
+    permissions:
+      contents: read
+      security-events: write
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Run Trivy vulnerability scanner
+        env:
+          TRIVY_DB_REPOSITORY: >-
+            ghcr.io/aquasecurity/trivy-db,public.ecr.aws/aquasecurity/trivy-db
+        uses: aquasecurity/trivy-action@0.28.0
+        with:
+          scan-type: "fs"
+          scan-ref: "."  # Default but being explicit
+          scanners: "vuln,secret"  # Default but being explicit
+          # ignore-unfixed: true
+          format: 'sarif'
+          output: 'trivy-results.sarif'
+          severity: 'CRITICAL,HIGH'
+          exit-code: "1"  # Fail workflow if vulnerabilities found
+
+      - name: Upload Trivy scan results to GitHub Security tab
+        if: always()  # Publish results to security tab even if scan fails
+        uses: github/codeql-action/upload-sarif@v3
```

Comment:
Although we still have one year before deprecation, since we have a warning already and the v4 has been released:
```suggestion
        uses: github/codeql-action/upload-sarif@v4
```